spec/services/course/user_invitation_service_spec.rb

```diff
@@ -0,0 +1,232 @@
+require 'rails_helper'
+
+RSpec.describe Course::UserInvitationService do
+  let(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    def temp_csv_from_attributes(records)
+      Tempfile.new(File.basename(__FILE__, '.*')).tap do |file|
+        file.write(CSV.generate do |csv|
+          csv << [:name, :email]
+          records.each do |user|
+            csv << [user.name, user.email]
+          end
+        end)
+        file.rewind
+      end
+    end
+
+    let(:course) { build(:course) }
+    subject { Course::UserInvitationService.new(course) }
+
+    let(:existing_users) do
+      (1..2).map do
+        create(:instance_user).user
+      end
+    end
+    let(:existing_user_attributes) do
+      existing_users.map do |user|
+        { name: user.name, email: user.email }
+      end
+    end
+    let(:new_users) do
+      (1..2).map do
+        create(:user)
+      end
+    end
+    let(:new_user_attributes) do
+      new_users.map do |user|
+        { name: user.name, email: user.email }
+      end
+    end
+    let(:users) { existing_users + new_users }
+    let(:user_attributes) { existing_user_attributes + new_user_attributes }
+
+    describe '#invite' do
+      def verify_new_user(user)
+        course_user = course.course_users.find do |course_user|
+          course_user.invitation && course_user.invitation.user_email.email == user.email
+        end
+        expect(course_user).not_to be_nil
+        expect(course_user).to be_invited
+        expect(course_user.name).to eq(user.name)
+      end
+
+      def verify_existing_user(user)
```

Comment:
Assignment Branch Condition size for verify_existing_user is too high. [19.05/15]